src/components/molecules/Visualizer/Layers/store.ts

```diff
@@ -68,6 +73,72 @@ export class LayerStore {
     return l;
   }
 
+  #usedIds: string[] = [];
+  #newId = () => {
+    const genResId = () =>
+      "_xxxxxxxxxxxxxxxxxxxxxxxxx".replace(/[x]/g, function () {
+        return ((Math.random() * 16) | 0).toString(16);
+      });
+    let id;
+    do {
+      id = genResId();
+    } while (this.#usedIds.includes(id));
+    this.#usedIds.push(id);
+    return id;
+  };
+
+  #insertLayer = (layerData: AddedLayerData, save = false) => {
+    const tar = layerData.parentId ? this.#map.get(layerData.parentId) : this.#root;
+    if (!tar?.children) return;
+    tar.children.push(layerData.layer);
+    const flattenedLayers = flattenLayers([layerData.layer]);
+    flattenedLayers.forEach(l => {
+      this.#map.set(l.id, l);
+      this.#pmap.set(l.id, this.#pluginLayer(l));
+    });
+    save && !tar.creator && this.#addedLayersData.push(layerData);
+    return layerData.layer.id;
+  };
+
+  #processLayer = (layer: Omit<Layer, "id">): Layer => {
+    const { children, ...self } = layer;
+    self.infobox?.blocks &&
+      self.infobox.blocks.forEach(b => {
+        b.id = this.#newId();
+      });
+    return {
+      ...self,
```

Comment:
```suggestion
    return {
      ...self,
      ...(self.infobox ? {
        infobox: {
          ...self.infobox,
          blocks: self.inbox.blocks?.map(b => ({ ...b, id: this.#newId() }))
        }
      } : {}),
```

It is safer to recreate the object unless performance is a concern, as changing the original object can lead to unexpected bugs. Also, the implementation is not complete if the value being returned is an object that has been rebuilt, but only the blocks have not been rebuilt.